test-data/unit/pythoneval.test

```diff
@@ -1458,3 +1458,51 @@ _testNoCrashOnGenericUnionUnpacking.py:10: error: Revealed type is 'Union[builti
 _testNoCrashOnGenericUnionUnpacking.py:11: error: Revealed type is 'Union[builtins.str, builtins.int]'
 _testNoCrashOnGenericUnionUnpacking.py:15: error: Revealed type is 'Union[builtins.int*, builtins.str*]'
 _testNoCrashOnGenericUnionUnpacking.py:16: error: Revealed type is 'Union[builtins.int*, builtins.str*]'
+
+[case testEnumIteration]
```

Comment:
> That's only part of what this is aiming to test.

Then choose a better name. Also the comment below is too long, just leave `# Regression test for #2305`

I am not arguing for fun here, there is `pytest -k` command that will run all test cases with a given pattern in names. This is why it is important to add some specific keyword like `Precise` or `Preserve` (or your choice). In addition, this avoids accidental test name collisions (we currently have thousands of test cases).